packages/vite/src/node/plugins/workerImportMetaUrl.ts

```diff
@@ -139,23 +177,35 @@ export function workerImportMetaUrlPlugin(config: ResolvedConfig): Plugin {
           let url: string
           if (isBuild) {
             const content = await bundleWorkerEntry(this, config, file)
-            const basename = path.parse(cleanUrl(file)).name
-            const contentHash = getAssetHash(content)
-            const fileName = path.posix.join(
-              config.build.assetsDir,
-              `${basename}.${contentHash}.js`
-            )
-            url = `__VITE_ASSET__${this.emitFile({
-              fileName,
-              type: 'asset',
-              source: content
-            })}__`
+            const inline = code.length < inlineLimit
+
+            if (inline) {
+              const workerOption = option.replace(/\s/g, '') || '{}'
+              // inline as blob data url
+              const inlineWorker = `(${inlineWorkerLoader.toString()})(${workerConstructor}, ${workerOption}, "${content.toString(
```

Comment:
`inlineWorkerLoader` if we can use `import from` expression we can have the better tree-shaking. but there will be got an html code we can't inject it in the head.

Reply:
place `webWorkerPlugin` plugin after `buildHtmlPlugin`